app/components/edition/details/fields/object_component.rb

```diff
@@ -1,16 +1,13 @@
 class Edition::Details::Fields::ObjectComponent < Edition::Details::Fields::BaseComponent
 private
 
-  def name_for_field(field)
-    field.name_attribute
-  end
-
-  def id_for_field(field)
-    field.id_attribute
-  end
-
-  def errors_for_field(field)
-    errors_for(edition.errors, field.error_key.to_sym)
+  def component_args(field)
+    {
+      edition:,
+      field:,
+      value: value_for_field(field),
+      schema:,
+    }
```

Comment:
nothing really to do with this refactoring but I think there's a unhelpful amount of indirection here with the `initialize` method being inherited from `Edition::Details::Fields::BaseComponent`, so `edition`, `field` and `schema` (and `@subschema` maybe?) are all implicit.

Would it be clearer if we used a module rather than inheritance to mix in the behaviour which we really want/need to share amongst the classes which are currently descendants of `Edition::Details::Fields::BaseComponent`? Am I right to think that what we want to share is (perhaps a subset of) the instance methods, and that it's actually not helpful to share the constructor which is then redefined in about half the descendants?